File: 백준/Silver/1620. 나는야 포켓몬 마스터 이다솜/나는야 포켓몬 마스터 이다솜.py
import sys
input = sys.stdin.readline
write = sys.stdout.write

def solve():
    line = input().split()
    if not line: return
    m, n = map(int, line)
    
    name_to_id = {} 
    id_to_name = [""] * (m + 1)
    
    for i in range(1, m + 1):
        pokemon = input().rstrip()
        name_to_id[pokemon] = i
        id_to_name[i] = pokemon
        
    result = []
    for _ in range(n):
        query = input().rstrip()
        if query.isdigit():
            result.append(id_to_name[int(query)])
        else:
            result.append(str(name_to_id[query]))
            
    write("\n".join(result) + "\n")

if __name__ == "__main__":
    solve()

File: 백준/Silver/1620. 나는야 포켓몬 마스터 이다솜/나는야 포켓몬 마스터 이다솜.cc
#include <iostream>
#include <vector>
#include <string>
#include <unordered_map>

using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int m = 0, n = 0;
    if (!(cin >> m >> n)) return 0;

    unordered_map<string, int> name_to_id;
    vector<string> id_to_name(m + 1);      

    for (int i = 1; i <= m; i++) {
        string name;
        if (!(cin >> name)) return 0;

        name_to_id[name] = i;
        id_to_name[i] = name;
    }

    for (int i = 0; i < n; i++) {
        string query;
        if (!(cin >> query)) return 0;

        if (isdigit(query[0])) {
            int id = stoi(query);
            cout << id_to_name[id] << "\n";
        }
        else {
            cout << name_to_id[query] << "\n";
        }
    }

    return 0;
}